### ISSUE TITLE
Confusing behavior when lang tags are used on non lang predicates

### ISSUE THREAD

**Issue by campoy** [dgraph-io/dgraph#4398]
## What version of Dgraph are you using?

master

### Have you tried reproducing the issue with the latest release?

yes

### What is the hardware spec (RAM, OS)?

n/a

### Steps to reproduce the issue (command/config used to run Dgraph).

Create a dataset with the following schema:

```rdf
<name>: string .
```

Send the following mutation:

```rdf
{
  set {
    _:a <name> "Foo" .
  }
}
```

#### Good

```graphql
{
  q(func: has(name)) {
    name@.
  }
}
```

Gives the (almost*) correct error message:

```
Message: : Language tags can only be used with predicates of string type having @lang directive in schema. Got: [name]
```

_*_ notice the extra `:`.

#### Good

```graphql
{
  q(func: has(name)) {
    name@*:.
  }
}
```

Gives the correct error message:

```
Message: If * is used, no other languages are allowed in the language list. Found [* .]
```

#### Bad

```graphql
{
  q(func: has(name)) {
    name@.:*
  }
}
```

Gives the error message, this doesn't make it clear that the problem is due to the star being used at the end.

```
Message: while lexing { q(func: has(name)) { name@.:* } } at line 3 column 11: Unrecognized character in lexText: U+002A '*'
```

#### Bad? Definitely weird

This query, asking for all of the languages shouldn't work. The same way, `name@.` doesn't.

```graphql
{
  q(func: has(name)) {
    name@*
  }
}
```

But it actually works and returns the values.

```json
{
  "data": {
    "q": [
      {
        "name": "Foo"
      }
    ]
  }}
```

**Comment by martinmr** [comment]
The lexing error is fixed by #4784. Now the language lists `*:.` and `.:*` return the same error.

The reason `name:*` returns a value because the logic says that it will return all the values, including those which have no language tag but `name:.` returns any value with a lang tag. I will work on a fix for this.

**Comment by martinmr** [comment]
Actually when the language tag is "." and there's no values with a language tag (but the predicate has language tags enabled), the query will return the value without the tag. So another solution would be to remove the restriction that the predicate has to have language enabled if the language list is `.`, in which case a value without a tag will be returned. This matches the current behavior of `name@*`.

**Comment by martinmr** [comment]
Fixed by #4881
